### PR TITLE
Habana DLC 1.4.1 Release

### PR DESCRIPTION
**Pull request by bukeao**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bukeao** [comment]
Note: [19ee4e7](https://github.com/aws/deep-learning-containers/pull/1900/commits/19ee4e7590172587355ea4ead4646c9e8874833b) is from Shantanu's comments on DLC 1.4.0 https://github.com/aws/deep-learning-containers/pull/1805

File: habana/tensorflow/buildspec-2-8.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
[the remark this answers is not in the document]
Comment (bukeao) [reply habana/tensorflow/buildspec-2-8.yml:1]: solved.

File: habana/pytorch/buildspec-1-10.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (shantanutrip) [inline habana/pytorch/buildspec-1-10.yml:1]: We need to follow the buildspec naming convention for this as well

Comment (shantanutrip) [reply habana/pytorch/buildspec-1-10.yml:1]: Also, when we do that, I guess we will have 2 buildspec files made, 1 for synai1.2 and the other for synai1.3.

File: habana/tensorflow/buildspec-2-8-syai-1-3.yml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (shantanutrip) [inline habana/tensorflow/buildspec-2-8-syai-1-3.yml:1]: We might also need to add the the older buildspec  for synAi 1.2 tf 2.7

**Review by shantanutrip — COMMENTED** [review]
I have provided a few comments. However, before addressing that, I would suggest that you merge the latest changes from  master into this branch and then address the buildpsec naming.

File: habana/tensorflow/buildspec-2-8-syai-1-3.yml (hunk 3 above)
Comment (bukeao) [reply habana/tensorflow/buildspec-2-8-syai-1-3.yml:1]: done

File: habana/pytorch/buildspec-1-10-syai-1-2.yml [hunk 4]
```diff
@@ -59,4 +59,4 @@ images:
     docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /example,
       /Dockerfile., *DEVICE_TYPE ]
     context:
-      <<: *TRAINING_CONTEXT
+      <<: *TRAINING_CONTEXT
```
Comment (shantanutrip) [inline habana/pytorch/buildspec-1-10-syai-1-2.yml:62]: Can we add new lines at the end of the files wherever required

File: dlc_developer_config.toml [hunk 5]
```diff
@@ -12,7 +12,7 @@ neuron_mode = false
 graviton_mode = false
```
Comment (shantanutrip) [inline dlc_developer_config.toml:12]: Lets revert changes in this file

File: test/dlc_tests/sanity/test_pre_release.py [hunk 6]
```diff
@@ -241,7 +241,7 @@ def test_framework_version_cpu(image):
         else:
             if "neuron" in image:
                 assert tag_framework_version in output.stdout.strip()
-            if all(_string in image for _string in ["pytorch", "habana", "synapseai1.3.0"]):
+            if all(_string in image for _string in ["pytorch", "habana", "synapseai1.4.1"]):
```
Comment (shantanutrip) [inline test/dlc_tests/sanity/test_pre_release.py:244]: In the interest of time, lets get this PR merged and then open a parallel PR to make this change backward compatible

File: dlc_developer_config.toml (hunk 5 above)
Comment (bukeao) [reply dlc_developer_config.toml:12]: done

**Review by shantanutrip — APPROVED** [review]
LGTM!! Once this is merged, lets open the parallel PR for making the change backward compatible.
